contracts/token/CRC1155/ICRC1155Enumerable.sol

```diff
@@ -0,0 +1,40 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts/token/ERC1155/IERC1155.sol";
+
+/**
+ * @dev Extension of ERC1155 to allow enumerability.
+ */
+interface ICRC1155Enumerable is IERC1155 {
+    /**
+     * @dev Returns the total amount of tokens stored by the contract.
+     */
+    function totalSupply() external view returns (uint256);
+
+    /**
+     * @dev Returns the `index`-th tokenId stored by the contract.
+     */
+    function tokenByIndex(uint256 index) external view returns (uint256);
+
+    /**
+     * @dev Returns the total amount of tokens for the specified `tokenId`.
+     */
+    function totalSupply(uint256 tokenId) external view returns (uint256);
+
+    /**
+     * @dev Indicates whether the specified `tokenId` exists or not.
+     */
+    function exists(uint256 tokenId) external view returns (bool);
+
+    /**
+     * @dev Returns the number of token ids held by `owner`.
+     */
+    function tokenCountOf(address owner) external view returns (uint256);
```

Comment:
Same situation here, it's not clear at first sight whether this would return the number of different token ids or the sum of the respective balances. I suppose it's the former.

Reply:
Yes, it is the total number of token ids. Generally, we use `amount` or `total supply` as the balance for a specified token id.

This API comes from a [pull request](https://github.com/OpenZeppelin/openzeppelin-contracts/pull/2707/files#diff-c3c93fccde6308a979504a26d6c33ef884f4be21a137b7015fa3e35592990521R21) of openzeppelin, which is closed due to expensive gas cost.